birdhouse/config/magpie/providers.cfg.template

```diff
@@ -62,6 +62,26 @@ providers:
     c4i: false
     type: thredds
     sync_type: thredds
+    configuration:
+      skip_prefix: "thredds"  # prefix to ignore, below prefixes will be matched against whatever comes after in path
+      file_patterns:
+        # note: make sure to employ quotes and double escapes to avoid parsing YAML error
+        - ".+\\.nc"
```

Comment:
Sorry, didn't bring this up earlier, did not yet had the chance to have a close look at this config file.

What is `file_patterns` exactly?  Effective for `data_type` only?  How is this different than the pattern (ex: `\\w+\\.ico`) in `metadata_type.prefixes`?

Reply:
description/example of all the parameters here: 
https://pavics-magpie.readthedocs.io/en/3.3.0/services.html#servicethredds

Reply:
```
# define which pattern matches that will map different path variations into same file resources
# this can be used to consider two file extensions as the same resource to avoid duplication of permissions
file_patterns:
```
Not very clear when reading this explanation.  But since we also have `.ncml` and `.txt` file on Thredds catalog, these should also be added there?

For "this can be used to consider two file extensions as the same resource to avoid duplication of permissions", can you add an example how to configure that to the doc?  I think Ouranos do not have that case yet (2 file extensions as same resource), do you have a live example of this case on some other Thredds server?

In fact, that doc will be very useful to refer back to, can you add a link to that doc directly in the config file so easy reference later?

Reply:
There is an example in the text below that sample config. 
Will add the link for reference.

Reply:
The general idea of that regex really is to map file/path to a same resource. 

So regex ` ".+\\.nc"` will match both `.nc` and `.ncml`. 
This means that browse/read/write permissions will apply to the same resource for both paths: 
`[...]/somedir/somefile.nc`
`[...]/somedir/somefile.ncml`

Without the regex, those two file would have 2 distinct Magpie resources, meaning that you would need to duplicate the permissions set each time to replicate access control over the same file for different "view formats".

When there are no regex matched, the filename are looked for literally, so the `.txt` don't need a regex.
A specific resource must be created if you want to set a premission specifically on it, but must of the time I think the parent directory with "recursive" will be used for that case (license file to its contents)?

Reply:
> So regex ` ".+\\.nc"` will match both `.nc` and `.ncml`.
> This means that browse/read/write permissions will apply to the same resource for both paths:
> `[...]/somedir/somefile.nc`
> `[...]/somedir/somefile.ncml`

For us the `.nc` and `.ncml` files are different file so different resource?  So new regex?  Here for example a `.ncml` file https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/catalog/datasets/gridded_obs/catalog.html?dataset=datasets/gridded_obs/nrcan_v2.ncml  So if you think it needs a new regex, please add it.

Reply:
In that case, yes it should be added to avoid truncating the extension to `.nc`. 
Added in f448778